core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GoogleServicesDAO.scala

```diff
@@ -232,17 +233,60 @@ abstract class GoogleServicesDAO(groupsPrefix: String) extends ErrorReportable {
 
   def deleteV1Project(googleProject: GoogleProjectId): Future[Unit]
 
+  def updateGoogleProject(googleProjectId: GoogleProjectId, googleProjectWithUpdates: Project): Future[Project]
+
   def deleteGoogleProject(googleProject: GoogleProjectId): Future[Unit]
 
   def getAccessTokenUsingJson(saKey: String): Future[String]
 
   def getUserInfoUsingJson(saKey: String): Future[UserInfo]
 
+  /**
+    * Convert a string to a legal gcp label text, with an optional prefix
+    * See: https://cloud.google.com/compute/docs/labeling-resources#restrictions
+    *
+    * @param s
+    * @param prefix defaults to "fc-"
+    * @return
+    */
   def labelSafeString(s: String, prefix: String = "fc-"): String = {
-    // https://cloud.google.com/compute/docs/labeling-resources#restrictions
     prefix + s.toLowerCase.replaceAll("[^a-z0-9\\-_]", "-").take(63)
   }
 
+  /**
+    * Convert a map of labels to legal gcp label text. Runs [[labelSafeString]] on all keys and values in the map.
+    * @param m Map of label key value pairs
+    * @param prefix
+    * @return
+    */
+  def labelSafeMap(m: Map[String, String], prefix: String = ""): Map[String, String] = m.map { case (key, value) =>
+    labelSafeString(key, prefix) -> labelSafeString(value, prefix) }
+
+  /**
+    * Valid text for google project name.
+    *
+    * "The optional user-assigned display name of the Project. It must be 4 to 30 characters. Allowed
+    * characters are: lowercase and uppercase letters, numbers, hyphen, single-quote, double-quote,
+    * space, and exclamation point."
+    *
+    * For more info see: https://cloud.google.com/resource-manager/reference/rest/v1/projects
+    * @param name
+    * @return
+    */
+  def googleProjectNameSafeString(name: String): String = {
+    name.replaceAll("[^a-zA-Z0-9\\-'\" !]", "-").take(30)
+  }
+
+  /**
+    * Handles getting the google project number from the google [[Project]]
+    * @param googleProject
+    * @return GoogleProjectNumber
+    */
+  def getGoogleProjectNumberFromGoogleProject(googleProject: Project): GoogleProjectNumber = googleProject.getProjectNumber match {
```

Comment:
nit: I think it's ok to call this just `getGoogleProjectNumber` and it will read a just as well